src/main/java/dev/galacticraft/mod/compat/emi/GalacticraftEmiPlugin.java

```diff
@@ -0,0 +1,230 @@
+                new MachineRecipeHandler<RecipeMachineMenu<SingleRecipeInput, SmeltingRecipe, ElectricFurnaceBlockEntity>>(
+                        ELECTRIC_SMELTING,
+                        List.of(ElectricFurnaceBlockEntity.INPUT_SLOT),
+                        ElectricFurnaceBlockEntity.OUTPUT_SLOT + 1
+                )
+        );
+        registry.addRecipeHandler(GCMenuTypes.ELECTRIC_ARC_FURNACE,
+                new MachineRecipeHandler<RecipeMachineMenu<SingleRecipeInput, BlastingRecipe, ElectricArcFurnaceBlockEntity>>(
+                        ELECTRIC_BLASTING,
+                        List.of(ElectricArcFurnaceBlockEntity.INPUT_SLOT),
+                        ElectricArcFurnaceBlockEntity.OUTPUT_SLOTS + ElectricArcFurnaceBlockEntity.OUTPUT_LENGTH
+                )
+        );
+        registry.addRecipeHandler(GCMenuTypes.ROCKET_WORKBENCH, new RocketRecipeHandler(ROCKET));
+
+        RecipeManager manager = registry.getRecipeManager();
+
+        for (RecipeHolder<FabricationRecipe> recipe : manager.getAllRecipesFor(GCRecipes.FABRICATION_TYPE)) {
+            registry.addRecipe(new FabricationEmiRecipe(recipe));
+        }
+
+        for (RecipeHolder<CompressingRecipe> recipe : manager.getAllRecipesFor(GCRecipes.COMPRESSING_TYPE)) {
+            registry.addRecipe(new CompressingEmiRecipe(recipe));
+            registry.addRecipe(new ElectricCompressingEmiRecipe(recipe));
+        }
+
+        for (RecipeHolder<SmeltingRecipe> recipe : manager.getAllRecipesFor(RecipeType.SMELTING)) {
+            registry.addRecipe(new ElectricSmeltingEmiRecipe(recipe));
+        }
+
+        for (RecipeHolder<BlastingRecipe> recipe : manager.getAllRecipesFor(RecipeType.BLASTING)) {
+            registry.addRecipe(new ElectricBlastingEmiRecipe(recipe));
+        }
+
+        for (RecipeHolder<RocketRecipe> recipe : manager.getAllRecipesFor(GCRecipes.ROCKET_TYPE)) {
+            registry.addRecipe(new RocketEmiRecipe(recipe));
+        }
+
+        registry.addRecipe(new EmiCraftingRecipe(
+                EmergencyKitItem.getContents(DyeColor.RED).stream()
+                        .map(itemStack -> itemStack.getItem() instanceof ParachuteItem
+                                ? EmiIngredient.of(GCItemTags.PARACHUTES)
+                                : EmiStack.of(itemStack))
+                        .collect(Collectors.toList()),
+                EmiStack.of(GCItems.EMERGENCY_KIT),
+                Constant.id("/" + Constant.Recipe.Serializer.EMERGENCY_KIT),
+                true
+        ));
+
+        addWorldInteraction(registry);
+    }
+
+    private static void addWorldInteraction(EmiRegistry registry) {
+        List<ItemStack> glassFluidPipes = new ArrayList<>();
+        for (Holder<Item> holder : BuiltInRegistries.ITEM.getTagOrEmpty(GCItemTags.GLASS_FLUID_PIPES)) {
+            glassFluidPipes.add(new ItemStack(holder));
+        }
+
+        ResourceLocation pipeInteraction = Constant.id("/world/glass_fluid_pipe/");
```

Comment:
Is it EMI convention to start identifiers with `/`-es? (I'm not familiar with the EMI api)